### ISSUE TITLE
Implementing new barostats? 

### ISSUE THREAD

**Issue by yummy-hat** [openmm/openmm#4238]
Hello,

I'd like to implement a barostat that doesn't exist in OpenMM, like potentially the Nose-Hoover method used in NAMD (described some [here](https://www.ks.uiuc.edu/Research/namd/2.9/ug/node37.html)). I understand that OpenMM only offers Monte Carlo barostats so calculation of the virial can be skipped, but I'm worried about how that will affect the dynamics of my system. 

Would this be possible through a CustomIntegrator as proposed in issue #635 (and maybe #2632)? Or are there plans to continue with options 3-5 in #635? 

I'm not super familiar with OpenMM internals, but it seems like calculation of the virial wouldn't be that hard: I was able to get decent efficiency using the StateDataReporter + finite difference method proposed in #3972 (available [here](https://github.com/z-gong/mstk/blob/master/mstk/ommhelper/reporter/statedatareporter.py#L470)). There has been plenty more discussion about other ways to compute the pressure (e.g. #3647, #1989, #3740, #147), so it seems like the interest is there. Plus, this would allow more direct comparison with GROMACS, LAMMPS, etc. in nice papers that implement new features like [this one](https://www.sciencedirect.com/science/article/pii/S0006349523002370).

Basically, I have three questions:
1. Are there plans to implement pressure reporting and/or other barostats? Is this a priority for developers?
2. Is it difficult to implement the virial calculation? Naively, it seems very straightforward. If performance is the issue, can't the computation just be made optional?
3. As someone with moderate Python/C++ experience and little familiarity with OpenMM internals, could I implement some other standard barostat using the tools in OpenMM now? And if possible, could someone provide some outline of what that might look like?

**Comment by peastman** [comment]
Computing a virial is quite complicated.  It requires changes to every routine that computes an energy throughout the whole codebase.  The idea gets discussed periodically, but it has never been a priority, especially considering the substantial burden it would create for code complexity and maintenance.

> Or are there plans to continue with options 3-5 in https://github.com/openmm/openmm/issues/635?

The different options discussed there were alternatives to each other.  We settled on MonteCarloMembraneBarostat as the best solution.
